craft_platforms/rock/_build.py

```diff
@@ -0,0 +1,42 @@
+# This file is part of craft-platforms.
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Rockcraft-specific platforms information."""
+
+from collections.abc import Sequence
+
+from craft_platforms import _buildinfo, _platforms
+
+
+def get_platforms_rock_build_plan(
```

Comment:
Since this is (currently) the only way to build a rock, we can probably just call it

```suggestion
def get_rock_build_plan(
```

Eventually we should have a `get_charm_build_plan` and a `get_snap_build_plan` that handle their various forms and are the correct functions to call.